File: CS-Examples/02_Data/ReplaceAndHighlight.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ReplaceAndHighlight
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load the workbook from file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ReplaceAndHighlight.xlsx");

            // Get the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Find all occurrences of the string "Total" in the worksheet, including case-sensitive and whole word matches
            CellRange[] ranges = worksheet.FindAllString("Total", true, true);

            // Iterate through each found range
            foreach (CellRange range in ranges)
            {
                // Reset the text in the range by replacing it with "Sum"
                range.Text = "Sum";

                // Set the color of the range to yellow
                range.Style.Color = Color.Yellow;
            }

            // Specify the file name for the resulting workbook after replacement and highlighting
            string result = "ReplaceAndHighlight_result.xlsx";

            // Save the modified workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Lauch the result file
            ExcelDocViewer(result);
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/02_Data/ReplaceAndHighlight.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace ReplaceAndHighlight
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ReplaceAndHighlight.xlsx")

            ' Get the first worksheet from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Find all occurrences of the string "Total" in the worksheet, case-sensitive and whole word matching
            Dim ranges() As CellRange = worksheet.FindAllString("Total", True, True)

            ' Iterate through each found range
            For Each range As CellRange In ranges
                ' Replace the text with "Sum"
                range.Text = "Sum"

                ' Set the color of the range to yellow
                range.Style.Color = Color.Yellow
            Next range

            ' Specify the output file name for the modified workbook
            Dim result As String = "ReplaceAndHighlight_result.xlsx"

            ' Save the workbook to the specified file path using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
